### ISSUE TITLE
[Modal] remove close icon toggle

### ISSUE THREAD

**Issue by laurenmrice** [carbon-design-system/carbon-design-kit#928]
## Acceptance criteria

- [x] Remove the toggle property to remove the close icon from Modals. According to a11y this option should not be available. It is not available in the code and has caused confusion in support (Ref [slack thread](https://ibm-studios.slack.com/archives/C0M053VPT/p1748354099359189)).

**Comment by mariuszmickiewicz** [comment]
* Removed the property
* Removed lock from the Close button for easier prototyping

Branch with the updates
https://www.figma.com/design/YAnB1jKx0yCUL29j6uSLpg/branch/boGDmtDxK3a9ZUCTepcQRM/-v11--Carbon-Design-System?m=auto&node-id=32639-353177&t=wiZxiYKCwfDQgs8u-1

**Comment by mariuszmickiewicz** [comment]
Branch merged to main.
